### ISSUE TITLE
can not define ID as uint64

### ISSUE THREAD

**Issue by TargetLiu** [facebook/ent#559]
https://github.com/facebookincubator/ent/blob/93c78aeda6e736b08da5a00a3ee8167854540af7/entc/gen/type.go#L768-L775

I want to define ID as uint64 without auto increment in order to use distributed id generation such as snowflake etc.

But according to the above code I cannot do this.

What is the reason why the type cannot be changed？

**Comment by a8m** [comment]
Hey @TargetLiu,
There's no real reason, I was created this way because we needed to support string/UUID IDs.
If we change it, we need to make sure we provide users an API to set the ID field to be incremented.

**Comment by TargetLiu** [comment]
@a8m So do you have plans to do this?

**Comment by a8m** [comment]
Hey @TargetLiu,
This issue can be addressed with additional field on the `entsql` annotation. See https://github.com/facebook/ent/pull/947 for a  reference. 

If you want to contribute this change, add another field `Increment *bool` to the `entsql.Annotation` and use it in [entc/type.go](https://github.com/facebook/ent/blob/master/entc/gen/type.go) as follows:

```go
func (f Field) PK() *schema.Column {
	c := &schema.Column{
		Name:      f.StorageKey(),
		Type:      f.Type.Type,
		Key:       schema.PrimaryKey,
		Increment: true,
	}
	if ant := f.EntSQL(); ant != nil && ant.Increment != nil {
		c.Increment = *ant.Increment
	}
	// ...
	return c
}
```

I suggested `*bool` for the field type, because `false` is also a valid value.

**Comment by saantiaguilera** [comment]
Hi! Coming from #1073 which has a similar issue. Was there any follow up on this thread? I can make a PR including this behavior without trouble.

On a side note, how's the release process for the module? Do you do it on demand based on new features or there's a schedule?

Thanks!!

**Comment by a8m** [comment]
> Hi! Coming from #1073 which has a similar issue. Was there any follow up on this thread? I can make a PR including this behavior without trouble.

Sure, let me know if you need some info/guidance and I can help with that (it will be tomorrow morning, because it's the EOD for me). 

> On a side note, how's the release process for the module? Do you do it on demand based on new features or there's a schedule?

No special process for this package, I'm releasing a new version when there are enough changes for a release or when someone asks for it in the issue-tracker or in [Slack](https://app.slack.com/client/T029RQSE6/C01FMSQDT53). But, users can always pull latest version using:
```
go get github.com/facebook/ent@master
```

**Comment by saantiaguilera** [comment]
@a8m I've opened https://github.com/facebook/ent/pull/1074 regarding this issue! Feel free to check it out!

**Comment by a8m** [comment]
Closed with #1074. Thank you both for your contribution 😃
